components/habitica/habitica.app.mjs

```diff
@@ -0,0 +1,147 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "app",
+  app: "habitica",
+  propDefinitions: {
+    group: {
+      type: "string",
+      label: "Group",
+      description: "ID of the groups to which the challenge belongs",
+      async options({ type }) {
+        const response = await this.getGroups({
+          type,
+        });
+        const groups = response.data;
+        return groups.map(({
+          name, _id,
+        }) => ({
+          label: name,
+          value: _id,
+        }));
+      },
+    },
+    name: {
+      type: "string",
+      label: "Name",
+      description: "Full name of the challenge",
+    },
+    shortName: {
+      type: "string",
+      label: "Short Name",
+      description: "A shortened name for the challenge, to be used as a tag",
+    },
+    summary: {
+      type: "string",
+      label: "Summary",
+      description: "A short summary advertising the main purpose of the challenge",
+      optional: true,
+    },
+    description: {
+      type: "string",
+      label: "Description",
+      description: "A detailed description of the challenge",
+      optional: true,
+    },
+    official: {
+      type: "boolean",
+      label: "Official",
+      description: "Whether or not a challenge is an official Habitica challenge",
+      optional: true,
+    },
+    prize: {
+      type: "string",
+      label: "Prize",
+      description: "Number of gems offered as a prize to challenge winne",
+      optional: true,
+    },
+    type: {
+      type: "string",
+      label: "Type",
+      description: "Type of the group",
+    },
```

Comment:
Should list options for `type`. From the [docs](https://habitica.com/apidoc/#api-Group-GetGroups): "Possible values are party, guilds, privateGuilds, publicGuilds, tavern."